assets/js/vue-apps/form-components/address-lookup.vue

```diff
@@ -0,0 +1,304 @@
+                    'CSRF-Token': token
+                },
+                data: { q: this.postcode }
+            })
+                .then(response => {
+                    if (response.addresses.length > 0) {
+                        this.currentState = this.states.Success;
+                        this.addressData = response.addresses;
+                        this.candidates = this.addressData.map(result => {
+                            const label = compact([
+                                result['line_1'],
+                                result['line_2'],
+                                result['post_town'],
+                                result['county']
+                            ]).join(', ');
+                            return { value: result.udprn, label: label };
+                        });
+                    } else {
+                        this.handleFailure();
+                    }
+                })
+                .fail(() => {
+                    this.handleFailure();
+                });
+        },
+        clearState() {
+            this.postcode = null;
+            this.candidates = [];
+            this.addressData = [];
+            this.fullAddress = null;
+        },
+        removeAddress() {
+            this.currentState = this.states.Asking;
+            this.clearState();
+            this.$emit('clear-address');
+        },
+        handleFallback() {
+            this.clearState();
+            this.$emit('show-fallback');
+        },
+        startEditing() {
+            this.currentState = this.states.Editing;
+            this.handleFallback();
+        },
+        updateAddressPreview(fullAddress) {
+            if (fullAddress) {
+                this.$emit('full-address', fullAddress);
+                this.currentState = this.states.AlreadyAnswered;
+                this.fullAddress = fullAddress;
+            }
+        },
+        trackFallbackState(fallbackVisibleState) {
+            this.fallbackVisible = fallbackVisibleState;
+        }
+    },
+    watch: {
+        selectedAddressId() {
+            if (this.selectedAddressId) {
+                this.currentState = this.states.Success;
+                const address = this.getAddressFromId(this.selectedAddressId);
+                if (address) {
+                    this.updateAddressPreview(this.formatAddress(address));
+                }
+            }
+        }
+    },
+    computed: {
+        elementIsHidden() {
+            return this.$el && this.$el.offsetParent === null;
+        },
+        shouldShowPostcodeLookup() {
+            return (
+                this.currentState === this.states.Asking ||
+                this.currentState === this.states.NotAsked ||
+                this.currentState === this.states.Success ||
+                this.currentState === this.states.Loading
+            );
+        },
+        formIsValid() {
+            const VALIDATION_REGEX = /^[a-z]{1,2}\d[a-z\d]?\s*\d[a-z]{2}$/i;
```

Comment:
Rather than inlining this regex you could use https://github.com/ideal-postcodes/postcode (which we already use for the `joi` postcode validation)